Sync/Synchronizers/Bookmarks/BookmarksSynchronizer.swift

```diff
@@ -229,12 +232,14 @@ open class BufferingBookmarksSynchronizer: TimestampedSingleCollectionSynchroniz
                 if case .completed = result {
                     return buffer.validate().bind { result in
                         if let invalidError = result.failureValue as? BufferInvalidError {
-                            self.statsSession.validationStats = self.validationStatsFrom(error: invalidError)
-
-                            log.warning("Buffer inconsistent, starting repair procedure")
-                            let repairer = BookmarksRepairRequestor(scratchpad: self.scratchpad, basePrefs: self.basePrefs, remoteClients: remoteClientsAndTabs)
-                            return repairer.startRepairs(validationInfo: invalidError.inconsistencies) >>> {
-                                deferMaybe(invalidError)
+                            return buffer.getUpstreamRecordCount() >>== { checked in
```

Comment:
Same.

Indeed, you might consider returning `Deferred<Int?>` instead of `Deferred<Result<Int>>`.